orion-server/src/main/java/com/pinterest/orion/core/actions/kafka/BrokerRecoveryAction.java

```diff
@@ -207,7 +208,15 @@ protected boolean checkServiceStabilityAfterRestart(String nodeId) {
 
   @Override
   public String getName() {
-    return "Broker Recovery" + (isDryRun ? " - Dry Run" : "");
+    // Different action names are required for BrokerRecoveryAction to be dispatched from same ClusterRecoveryAction.
+    String nodeId = "Unknown Node";
+    if (containsAttribute(OrionConstants.NODE_ID)) {
+      nodeId = getAttribute(OrionConstants.NODE_ID).getValue();
+    }
+    return String.format(
+            "BrokerRecoveryAction for %s %s",
+           nodeId,
+            (isDryRun ? " - Dry Run" : ""));
```

Comment:
What's the use case for dry run?

Reply:
When dryrun is true and the action is triggerd, the alarm will be triggered but the EC2 related action (restart/replace) will not be triggerred. It's legacy code used for testing purpose.